packages/x-date-pickers-pro/src/internals/hooks/useStaticRangePicker/useStaticRangePicker.tsx

```diff
@@ -36,55 +36,44 @@ export const useStaticRangePicker = <
 }: UseStaticRangePickerParams<TView, TExternalProps>) => {
   const { localeText, slots, slotProps, className, sx, displayStaticWrapperAs, autoFocus } = props;
 
-  const { rangePosition, onRangePositionChange } = useRangePosition(props);
+  const rangePositionResponse = useRangePosition(props);
 
   const { layoutProps, providerProps, renderCurrentView } = usePicker<
     PickerRangeValue,
     TView,
-    TExternalProps,
-    {}
+    TExternalProps
   >({
     ...pickerParams,
     props,
     autoFocusView: autoFocus ?? false,
     fieldRef: undefined,
     localeText,
-    additionalViewProps: {
-      rangePosition,
-      onRangePositionChange,
-    },
     variant: displayStaticWrapperAs,
   });
 
   const Layout = slots?.layout ?? PickerStaticLayout;
-  const slotPropsForLayout: PickersLayoutSlotProps<PickerRangeValue> = {
-    ...slotProps,
-    toolbar: {
-      ...slotProps?.toolbar,
-      rangePosition,
-      onRangePositionChange,
-    } as ExportedBaseToolbarProps,
-  };
 
   const renderPicker = () => (
-    <PickerProvider {...providerProps}>
-      <Layout
-        {...layoutProps}
-        {...slotProps?.layout}
-        slots={slots}
-        slotProps={slotPropsForLayout}
-        sx={[
-          ...(Array.isArray(sx) ? sx : [sx]),
-          ...(Array.isArray(slotProps?.layout?.sx)
-            ? slotProps!.layout!.sx
-            : [slotProps?.layout?.sx]),
-        ]}
-        className={clsx(className, slotProps?.layout?.className)}
-        ref={ref}
-      >
-        {renderCurrentView()}
-      </Layout>
-    </PickerProvider>
+    <PickerRangePositionContext.Provider value={rangePositionResponse}>
+      <PickerProvider {...providerProps}>
+        <Layout
+          {...layoutProps}
+          {...slotProps?.layout}
+          slots={slots}
+          slotProps={slotProps}
+          sx={[
+            ...(Array.isArray(sx) ? sx : [sx]),
+            ...(Array.isArray(slotProps?.layout?.sx)
```

Comment:
We could introduce a helper function for this that wraps non array values into a single entry array 🤔

Reply:
Indeed, it could be an interesting issue for a small refactoring

Reply:
I know. It just feels a bit clunky and repetitive